### ISSUE TITLE
Not able to chat with agents when using Azure model provider because of not being able to select the the correct target model(values provided in dropdown are incorrect)

### ISSUE THREAD

**Issue by sangee2004** [otto8-ai/otto8#702]
**Pre req** - Have the following Azure Environment variables set:

AZURE_OPENAI_MODEL_PROVIDER_API_KEY , OTTO8_AZURE_OPENAI_MODEL_PROVIDER_DEPLOYMENT_NAME, OTTO8_AZURE_OPENAI_MODEL_PROVIDER_ENDPOINT

**Steps to reproduce the problem:**
1. Add a model using `Add Model` option and select "Azure OpenAI" as model provider.
2. Notice that the options provided in the model dropdown list are incorrect (they are the same list as the one presented for OpenAI) 

<img width="607" alt="Screenshot 2024-11-26 at 2 38 28 PM" src="https://github.com/user-attachments/assets/ce080162-ea7a-4378-8e3e-53ac0d9be084">

3. Picked `gpt-40` option from this list and created a model provider. 
4. Made this the default LLM provider.

Chatting with agents results in the following error:
```
failed to run: failed calling model for completion: error, status code: 404, message: {"detail":"Error occurred: Error code: 404 - {'error': {'code': 'DeploymentNotFound', 'message': 'The API deployment for this resource does not exist. If you created the deployment within the last 5 minutes, please wait a moment and try again.'}}"}
```

**Expected Behavior:**
User should be presented with the correct model provider list that is available for the configured Azure OpenAI configuration.

**Comment by cjellick** [comment]
assigned to @thedadams for the backend, @ryanhopperlowe for the frontend

**Comment by ryanhopperlowe** [comment]
@cjellick - This should be fixed with #744
